src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs

```diff
@@ -119,22 +175,34 @@ public Task<(object Snapshot, long Index)> GetSnapshotAsync(string actorName)
                 throw ex;
             }
 
-            return Task.FromResult((snapshot, index));
+            return (snapshot, index);
         }
 
         public async Task<long> PersistEventAsync(string actorName, long index, object @event)
         {
             try
             {
-                using (var db = new SqlitePersistenceContext(_datasource))
-                {
-                    var item = new Event(actorName, index, JsonConvert.SerializeObject(@event, new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.All }));
-
-                    await db.Events.AddAsync(item);
+                var item = new Event(actorName, index, JsonConvert.SerializeObject(@event, new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.All }));
 
-                    await db.SaveChangesAsync();
+                using (var connection = new SqliteConnection(_connectionString))
+                {
+                    connection.Open();
 
-                    return index++;
+                    using (var transaction = connection.BeginTransaction())
```

Comment:
Don't need a transaction over a single command.  Sqlite implicitly has a transaction on this.